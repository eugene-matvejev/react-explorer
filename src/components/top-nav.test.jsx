import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import TopNav from './top-nav';

configure({ adapter: new Adapter() });

describe('<TopNav/>', () => {
    const props = {
        label: '{{label}}',
    };
    describe('render', () => {
        it('with default/require d props', () => {
            const c = shallow(<TopNav {...props} />);

            expect(c).toMatchSnapshot();
        });

        describe('with optional props', () => {
            [
                ['className', '{{className}}'],
                ['data-cy', '{{data-cy}}'], /** should be passed 'as is' */
            ].forEach(([prop, v]) => {
                it(`[::${prop}] as "${v}"`, () => {
                    const c = shallow(<TopNav {...props} {...{ [prop]: v }} />);

                    expect(c).toMatchSnapshot();
                });
            });
        });
    });
});

