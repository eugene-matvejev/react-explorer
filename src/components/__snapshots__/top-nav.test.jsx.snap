// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopNav/> render with default/require d props 1`] = `
<nav
  className="topnav "
  label="{{label}}"
>
  <img
    alt="logo"
    className="topnav__logo"
    src="/assets/img/placeholder.png"
  />
</nav>
`;

exports[`<TopNav/> render with optional props [::className] as "{{className}}" 1`] = `
<nav
  className="topnav {{className}}"
  label="{{label}}"
>
  <img
    alt="logo"
    className="topnav__logo"
    src="/assets/img/placeholder.png"
  />
</nav>
`;

exports[`<TopNav/> render with optional props [::data-cy] as "{{data-cy}}" 1`] = `
<nav
  className="topnav "
  data-cy="{{data-cy}}"
  label="{{label}}"
>
  <img
    alt="logo"
    className="topnav__logo"
    src="/assets/img/placeholder.png"
  />
</nav>
`;
